lifecycle-operator/controllers/lifecycle/keptnappversion/controller_test.go

```diff
@@ -117,6 +117,71 @@ func TestKeptnAppVersionReconciler_reconcile(t *testing.T) {
 
 }
 
+func TestKeptnAppVersionNonBlockingReconciler_Reconcile(t *testing.T) {
+
+	status := apilifecycle.KeptnAppVersionStatus{
```

Comment:
Thank you for the contribution, but correct me if I am wrong, this test reconciles already ended reconciliation of the KeptnAppVersion, therefore no action is perfomed. I would expect a test where there is a change performed on the custom resource. Can you please explain the test briefly?

Thank you!

Reply:
Ah! You're absolutely right! I'd already reconciled the object before it reconciles!

After a little more digging around, I came up with an idea. 

I think only `PreDeploymentTaskStatus` needs to be maintained to `StateFailed`. For this test, we can have the reconciler figure out the status of the `PreDeploymentStatus` field after having the `MockHandler` reconcile the phase. We can then test whether `PreDeploymentStatus` comes out to be `StateWarning`. A blocking controller would result in this Status becoming `StateFailed`.  In this case, since we are using a blocking controller, the Status would become `StateWarning`.

Am I understanding this correctly? 

Since we are using a mock PhaseHandler, do we need to develop a